src/wechaty_puppet_itchat/puppet.py

```diff
@@ -0,0 +1,1098 @@
+    get_endpoint,
+    get_token,
+)
+
+log = get_logger('HostiePuppet')
+
+
+def _map_message_type(message_payload: MessagePayload) -> MessagePayload:
+    """
+    get messageType value which is ts-wechaty-puppet type from service server,
+        but is MessageType. so we should map it to MessageType from wechaty-grpc
+    target MessageType Enum:
+        MESSAGE_TYPE_UNSPECIFIED  = 0;
+
+       MESSAGE_TYPE_ATTACHMENT   = 1;
+       MESSAGE_TYPE_AUDIO        = 2;
+       MESSAGE_TYPE_CONTACT      = 3;
+       MESSAGE_TYPE_EMOTICON     = 4;
+       MESSAGE_TYPE_IMAGE        = 5;
+       MESSAGE_TYPE_TEXT         = 6;
+       MESSAGE_TYPE_VIDEO        = 7;
+       MESSAGE_TYPE_CHAT_HISTORY = 8;
+       MESSAGE_TYPE_LOCATION     = 9;
+       MESSAGE_TYPE_MINI_PROGRAM = 10;
+       MESSAGE_TYPE_TRANSFER     = 11;
+       MESSAGE_TYPE_RED_ENVELOPE = 12;
+       MESSAGE_TYPE_RECALLED     = 13;
+       MESSAGE_TYPE_URL          = 14;
+
+    source MessageType Enum:
+        export enum MessageType {
+          Unknown = 0,
+
+          Attachment=1,     // Attach(6),
+          Audio=2,          // Audio(1), Voice(34)
+          Contact=3,        // ShareCard(42)
+          ChatHistory=4,    // ChatHistory(19)
+          Emoticon=5,       // Sticker: Emoticon(15), Emoticon(47)
+          Image=6,          // Img(2), Image(3)
+          Text=7,           // Text(1)
+          Location=8,       // Location(48)
+          MiniProgram=9,    // MiniProgram(33)
+          GroupNote=10,      // GroupNote(53)
+          Transfer=11,       // Transfers(2000)
+          RedEnvelope=12,    // RedEnvelopes(2001)
+          Recalled=13,       // Recalled(10002)
+          Url=14,            // Url(5)
+          Video=15,          // Video(4), Video(43)
+        }
+    :return:
+
+    #
+    """
+    if isinstance(message_payload.type, int):
+        map_container: List[MessageType] = [
+            MessageType.MESSAGE_TYPE_UNSPECIFIED,
+            MessageType.MESSAGE_TYPE_ATTACHMENT,
+            MessageType.MESSAGE_TYPE_AUDIO,
+            MessageType.MESSAGE_TYPE_CONTACT,
+            MessageType.MESSAGE_TYPE_CHAT_HISTORY,
+            MessageType.MESSAGE_TYPE_EMOTICON,
+            MessageType.MESSAGE_TYPE_IMAGE,
+            MessageType.MESSAGE_TYPE_TEXT,
+            MessageType.MESSAGE_TYPE_LOCATION,
+            MessageType.MESSAGE_TYPE_MINI_PROGRAM,
+            MessageType.MESSAGE_TYPE_UNSPECIFIED,
+            MessageType.MESSAGE_TYPE_TRANSFER,
+            MessageType.MESSAGE_TYPE_RED_ENVELOPE,
+            MessageType.MESSAGE_TYPE_RECALLED,
+            MessageType.MESSAGE_TYPE_URL,
+            MessageType.MESSAGE_TYPE_VIDEO]
+        message_payload.type = map_container[message_payload.type]
+    return message_payload
+
+
+# pylint: disable=R0904
+class PuppetItChat(Puppet):
+    """
+    grpc wechaty puppet implementation
+    """
+
+    def __init__(self, options: PuppetOptions, name: str = 'puppet_service'):
```

Comment:
A ha, this name should be `puppet-itchat`.

Reply:
BTW, there are a lot `puppet service` in this module. You should read the code to refactor every method.